openpype/hosts/maya/api/pipeline.py

```diff
@@ -426,6 +478,47 @@ def on_before_save():
     return lib.validate_fps()
 
 
+def check_lock_on_current_file():
+
+    """Check if there is a user opening the file"""
+    if not handle_workfile_locks():
+        return
+    log.info("Running callback on checking the lock file...")
+
+    # add the lock file when opening the file
+    filepath = current_file()
+
+    if not is_workfile_locked(filepath):
+        create_workfile_lock(filepath)
+        return
+
+    # add lockfile dialog
+    from Qt import QtWidgets
+    from openpype.tools.workfiles.lock_dialog import WorkfileLockDialog
+
+    top_level_widgets = {w.objectName(): w for w in
+                         QtWidgets.QApplication.topLevelWidgets()}
+    parent = top_level_widgets.get("MayaWindow", None)
+    workfile_dialog = WorkfileLockDialog(filepath, parent=parent)
+    workfile_dialog.show()
```

Comment:
Technically the lock could have been removed between "is_workfile_locked" and initializing the WorkfileLockDialog.
Would it make sense more to instead do it like this:

```python
lockfile_data = get_lockfile_data(filepath)
if not lockfile_data:
    # File is not locked, create lock, etc.
    return
 
dialog = WorkfileLockDialog(lock_data, parent=parent)
dialog.show()
```
Otherwise one could run into the race condition where the WorkfileLockDialog would be unable to get the data and then raise an error doing so?